[Bug] Crashes

### Describe the bug

When I try to launch my game crashes.

### Crashlog

---- Minecraft Crash Report ----
// This doesn't make any sense!

Time: 2026-03-25 21:59:03
Description: Initializing game

java.lang.NoClassDefFoundError: com/nikoverflow/exploitpreventer/module/Modules
	at knot//meteordevelopment.meteorclient.systems.modules.misc.ServerSpoof.md2aaa0d$meteor-rejects$lambda$onInit$0$2(ServerSpoof.java:536)
	at knot//meteordevelopment.meteorclient.settings.Setting.onChanged(Setting.java:92)
	at knot//meteordevelopment.meteorclient.settings.Setting.reset(Setting.java:67)
	at knot//meteordevelopment.meteorclient.systems.modules.Modules.load(Modules.java:87)
	at knot//meteordevelopment.meteorclient.systems.Systems.load(Systems.java:92)
	at knot//meteordevelopment.meteorclient.systems.Systems.load(Systems.java:98)
	at knot//meteordevelopment.meteorclient.MeteorClient.onInitializeClient(MeteorClient.java:137)
	at knot//net.minecraft.class_310.handler$zgj005$meteor-client$onInit(class_310.java:3703)
	at knot//net.minecraft.class_310.<init>(class_310.java:745)
	at knot//net.minecraft.client.main.Main.main(Main.java:234)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:514)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:72)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.modrinth.theseus.MinecraftLaunch.relaunch(MinecraftLaunch.java:63)
	at com.modrinth.theseus.MinecraftLaunch.main(MinecraftLaunch.java:28)
Caused by: java.lang.ClassNotFoundException: com.nikoverflow.exploitpreventer.module.Modules
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:239)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 17 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at knot//meteordevelopment.meteorclient.systems.modules.misc.ServerSpoof.md2aaa0d$meteor-rejects$lambda$onInit$0$2(ServerSpoof.java:536)
	at knot//meteordevelopment.meteorclient.settings.Setting.onChanged(Setting.java:92)
	at knot//meteordevelopment.meteorclient.settings.Setting.reset(Setting.java:67)
	at knot//meteordevelopment.meteorclient.systems.modules.Modules.load(Modules.java:87)
	at knot//meteordevelopment.meteorclient.systems.Systems.load(Systems.java:92)
	at knot//meteordevelopment.meteorclient.systems.Systems.load(Systems.java:98)
	at knot//meteordevelopment.meteorclient.MeteorClient.onInitializeClient(MeteorClient.java:137)
	at knot//net.minecraft.class_310.handler$zgj005$meteor-client$onInit(class_310.java:3703)
	at knot//net.minecraft.class_310.<init>(class_310.java:745)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.26100.3624 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:User Experience Controls Library:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.26100.7623 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.26100.7705 (WinBuild.160101.0800):Microsoft Corporation
		CoreMessaging.dll:Microsoft CoreMessaging Dll:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		CoreUIComponents.dll:Microsoft Core UI Components Dll:10.0.26100.7627:Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.26100.5074 (WinBuild.160101.0800):Microsoft Corporation
		DEVOBJ.dll:Device Information Set DLL:10.0.26100.1150 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.26100.1591 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.26100.8036 (WinBuild.160101.0800):Microsoft Corporation
		GLU32.dll:OpenGL Utility Library DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.26100.7309 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		MMDevApi.dll:MMDevice API:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		MSASN1.dll:ASN.1 Runtime APIs:10.0.26100.7623 (WinBuild.160101.0800):Microsoft Corporation
		MSCTF.dll:MSCTF Server DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		MpOav.dll:IOfficeAntiVirus Module:4.18.26010.5 (1e74b9adf5ed0dd9a81fac912007d1b65cd9b869):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		OLEAUT32.dll:OLEAUT32.DLL:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.26100.6725 (WinBuild.160101.0800):Microsoft Corporation
		OpenAL.dll:Main implementation library:1.23.1:
		POWRPROF.dll:Power Profile Helper DLL:10.0.26100.4202 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		SETUPAPI.dll:Windows Setup API:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Windows Shell Common Dll:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		UMPDC.dll:User Mode Power Dependency Coordinator:10.0.26100.7019 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Multi-User Windows USER API Client DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.40.33810.0:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.26100.1150 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.26100.5074 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		WINTRUST.dll:Microsoft Trust Verification APIs:10.0.26100.8036 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		amdihk64.dll:Radeon Settings: Host Service:2,00,00,1788:Advanced Micro Devices, Inc.
		amsi.dll:Anti-Malware Scan Interface:10.0.26100.7309 (WinBuild.160101.0800):Microsoft Corporation
		apphelp.dll:Application Compatibility Client Library:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		atig6pxx.dll:atiglpxx.dll:8.14.01.6564:Advanced Micro Devices, Inc. 
		atio6axx.dll:AMD OpenGL Driver:26.03.250928_8576ea2:Advanced Micro Devices, Inc
		awt.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		bcrypt.dll:Windows Cryptographic Primitives Library:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.26100.7623 (WinBuild.160101.0800):Microsoft Corporation
		cfgmgr32.dll:Configuration Manager DLL:10.0.26100.7309 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM for Windows:10.0.26100.7705 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		dinput8.dll:Microsoft DirectInput:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		dwmapi.dll:Microsoft Desktop Window Manager API:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		dxcore.dll:DXCore:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		extnet.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		fastprox.dll:WMI Custom Marshaller:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		fwpuclnt.dll:FWP/IPsec User-Mode API:10.0.26100.3915 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.26100.8036 (WinBuild.160101.0800):Microsoft Corporation
		glfw.dll:GLFW 3.4.0 DLL:3.4.0:GLFW
		icm32.dll:Microsoft Color Management Module (CMM):10.0.26100.2314 (WinBuild.160101.0800):Microsoft Corporation
		inputhost.dll:InputHost:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		instrument.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		java.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		javaw.exe:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		jemalloc.dll
		jimage.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		jli.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		jna7782904931604974131.dll:JNA native library:7.0.2:Java(TM) Native Access (JNA)
		jsvml.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		jvm.dll:Zulu 64-Bit Server VM:21.0.10:Azul Systems Inc.
		kernel.appcore.dll:AppModel API Host:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		lwjgl_opengl.dll
		lwjgl_stb.dll
		management.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		management_ext.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		mscms.dll:Microsoft Color Matching System DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		msvcp140.dll:Microsoft® C Runtime Library:14.40.33810.0:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.26100.7623 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.26100.7623 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0 Service Provider:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-mail Naming Shim Provider:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt Router:10.0.26100.1591 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		nio.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		nlansp_c.dll:NLA Namespace Service Provider DLL:10.0.26100.7309 (WinBuild.160101.0800):Microsoft Corporation
		ntdll.dll:NT Layer DLL:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		opengl32.dll:OpenGL Client DLL:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:Windows System Performance Objects DLL:10.0.26100.7705 (WinBuild.160101.0800):Microsoft Corporation
		pfclient.dll:SysMain Client:10.0.26100.4202 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.26100.7309 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		sapi.dll:Speech API:5.3.29131.00 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight Utility Library:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		textinputframework.dll:"TextInputFramework.DYNLINK":10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.26100.7623 (WinBuild.160101.0800):Microsoft Corporation
		uxtheme.dll:Microsoft UxTheme Library:10.0.26100.7019 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.40.33810.0:Microsoft Corporation
		verify.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
		wbemcomn.dll:WMI:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		wbemprox.dll:WMI:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		wbemsvc.dll:WMI:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		win32u.dll:Win32u:10.0.26100.8036 (WinBuild.160101.0800):Microsoft Corporation
		windows.staterepositorycore.dll:Windows StateRepository API Core:10.0.26100.7920 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage API:10.0.26100.7840 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.26100.1882 (WinBuild.160101.0800):Microsoft Corporation
		wintypes.dll:Windows Base Types DLL:10.0.26100.7705 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.26100.5074 (WinBuild.160101.0800):Microsoft Corporation
		wshunix.dll:AF_UNIX Winsock2 Helper DLL:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		xinput1_4.dll:Microsoft Common Controller API:10.0.26100.1 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:Zulu Platform x64 Architecture:21.0.10:Azul Systems Inc.
Stacktrace:
	at knot//net.minecraft.client.main.Main.main(Main.java:234)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:514)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:72)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.modrinth.theseus.MinecraftLaunch.relaunch(MinecraftLaunch.java:63)
	at com.modrinth.theseus.MinecraftLaunch.main(MinecraftLaunch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.21.11
	Minecraft Version ID: 1.21.11
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 21.0.10, Azul Systems, Inc.
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Azul Systems, Inc.
	Memory: 102327792 bytes (97 MiB) / 710934528 bytes (678 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 16
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 7 7800X3D 8-Core Processor           
	Identifier: AuthenticAMD Family 25 Model 97 Stepping 2
	Microarchitecture: Zen 3
	Frequency (GHz): 4.20
	Number of physical packages: 1
	Number of physical CPUs: 8
	Number of logical CPUs: 16
	Graphics card #0 name: AMD Radeon RX 9070 XT
	Graphics card #0 vendor: Advanced Micro Devices, Inc.
	Graphics card #0 VRAM (MiB): 16304.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 32.0.23033.1002
	Graphics card #1 name: AMD Radeon(TM) Graphics
	Graphics card #1 vendor: Advanced Micro Devices, Inc.
	Graphics card #1 VRAM (MiB): 512.00
	Graphics card #1 deviceId: VideoController2
	Graphics card #1 versionInfo: 32.0.21043.5001
	Memory slot #0 capacity (MiB): 16384.00
	Memory slot #0 clockSpeed (GHz): 4.80
	Memory slot #0 type: DDR5
	Memory slot #1 capacity (MiB): 16384.00
	Memory slot #1 clockSpeed (GHz): 4.80
	Memory slot #1 type: DDR5
	Virtual memory max (MiB): 33958.32
	Virtual memory used (MiB): 17723.70
	Swap memory total (MiB): 2048.00
	Swap memory used (MiB): 12.03
	Space in storage for jna.tmpdir (MiB): available: 766143.88, total: 975802.00
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): available: 766143.88, total: 975802.00
	Space in storage for io.netty.native.workdir (MiB): available: 766143.88, total: 975802.00
	Space in storage for java.io.tmpdir (MiB): available: 766143.88, total: 975802.00
	Space in storage for workdir (MiB): available: 766143.88, total: 975802.00
	JVM Flags: 2 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2048M
	Debug Flags: 0 total; 
	Launched Version: 1.21.11
	Launcher name: theseus
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: AMD Radeon RX 9070 XT GL version 3.3.0 Core Profile Context 26.3.1.260309, ATI Technologies Inc.
	Window size: <not initialized>
	GFLW Platform: win32
	Render Extensions: GL_ARB_buffer_storage, GL_KHR_debug, GL_ARB_vertex_attrib_binding, GL_ARB_direct_state_access, GL_EXT_texture_filter_anisotropic
	GL debug messages: 
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 404
	Type: Client (map_client.txt)
	Locale: en_US
	System encoding: Cp1252
	File encoding: UTF-8
	CPU: 16x AMD Ryzen 7 7800X3D 8-Core Processor 

-- Meteor Client --

Version: 1.21.11
Build: 73


### Steps to reproduce

1.
2.
3.
...


### Operating System

None

### Addon Version

_No response_

### Meteor Version

1.21.11 #73

### Minecraft Version

1.21.11

### Other mods

```plaintext
No other mods
```

--- a/build.gradle
+++ b/build.gradle
@@ -62,8 +62,8 @@ dependencies {
 	implementation (include('com.github.19MisterX98.SeedcrackerX:seedcrackerx-api:2.10.1')) {transitive = false}
 //	implementation (include('com.github.19MisterX98.SeedcrackerX:seedcrackerx-api:master-SNAPSHOT')) {transitive = false}
 
-	// Exploit Preventer
-	modImplementation("com.nikoverflow:exploit-preventer-api:0.0.1")
+	// Exploit Preventer (optional)
+	modCompileOnly("com.nikoverflow:exploit-preventer-api:0.0.1")
 
 	configurations.implementation.extendsFrom(configurations.extraLibs)
 }

--- a/src/main/java/anticope/rejects/mixin/meteor/modules/ServerSpoofMixin.java
+++ b/src/main/java/anticope/rejects/mixin/meteor/modules/ServerSpoofMixin.java
@@ -1,13 +1,13 @@
 package anticope.rejects.mixin.meteor.modules;
 
-import com.nikoverflow.exploitpreventer.ExploitPreventer;
-import com.nikoverflow.exploitpreventer.module.Modules;
+import anticope.rejects.utils.ExploitPreventerCompat;
 import meteordevelopment.meteorclient.settings.BoolSetting;
 import meteordevelopment.meteorclient.settings.Setting;
 import meteordevelopment.meteorclient.settings.SettingGroup;
 import meteordevelopment.meteorclient.systems.modules.Category;
 import meteordevelopment.meteorclient.systems.modules.Module;
 import meteordevelopment.meteorclient.systems.modules.misc.ServerSpoof;
+import net.fabricmc.loader.api.FabricLoader;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
@@ -15,6 +15,7 @@ import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(value = ServerSpoof.class, remap = false)
 public class ServerSpoofMixin extends Module {
+    private static final boolean EP_LOADED = FabricLoader.getInstance().isModLoaded("exploitpreventer");
 
     private SettingGroup sgExploitPreventer;
     private Setting<Boolean> translationKey;
@@ -33,7 +34,10 @@ public class ServerSpoofMixin extends Module {
             .name("translation-key-prevention")
             .description("Prevents the server from detecting installed mods via translation keys.")
             .defaultValue(false)
-            .onChanged(val -> applyModule(Modules.TRANSLATION_KEY, val))
+            .onChanged(val -> {
+                if (!EP_LOADED) { if (val) warning("ExploitPreventer is not installed."); return; }
+                ExploitPreventerCompat.applyTranslationKey(val);
+            })
             .build()
         );
 
@@ -41,7 +45,10 @@ public class ServerSpoofMixin extends Module {
             .name("fingerprint-prevention")
             .description("Stops servers from using resource packs to uniquely identify your client.")
             .defaultValue(false)
-            .onChanged(val -> applyModule(Modules.FINGERPRINTING, val))
+            .onChanged(val -> {
+                if (!EP_LOADED) { if (val) warning("ExploitPreventer is not installed."); return; }
+                ExploitPreventerCompat.applyFingerprinting(val);
+            })
             .build()
         );
 
@@ -49,27 +56,23 @@ public class ServerSpoofMixin extends Module {
             .name("local-HTTP-request-prevention")
             .description("Blocks resource pack URLs that point to local IP addresses, prevents detection of locally running services.")
             .defaultValue(false)
-            .onChanged(val -> applyModule(Modules.LOCAL_HTTP_REQUEST, val))
+            .onChanged(val -> {
+                if (!EP_LOADED) { if (val) warning("ExploitPreventer is not installed."); return; }
+                ExploitPreventerCompat.applyLocalHTTPRequest(val);
+            })
             .build()
         );
     }
 
     @Inject(method = "onActivate", at = @At("TAIL"))
     private void onActivate(CallbackInfo ci) {
-        applyModule(Modules.TRANSLATION_KEY, translationKey.get());
-        applyModule(Modules.FINGERPRINTING, fingerprinting.get());
-        applyModule(Modules.LOCAL_HTTP_REQUEST, localHTTPRequest.get());
+        if (!EP_LOADED) return;
+        ExploitPreventerCompat.applyAll(translationKey.get(), fingerprinting.get(), localHTTPRequest.get());
     }
 
-    @Inject(method = "onDeactivate", at=@At("TAIL"), remap = false)
+    @Inject(method = "onDeactivate", at = @At("TAIL"), remap = false)
     private void onDeactivate(CallbackInfo ci) {
-        applyModule(Modules.TRANSLATION_KEY, false);
-        applyModule(Modules.FINGERPRINTING, false);
-        applyModule(Modules.LOCAL_HTTP_REQUEST, false);
-    }
-
-    private void applyModule(Modules module, boolean enabled) {
-        if (ExploitPreventer.getAPI() == null) return;
-        ExploitPreventer.getAPI().getModuleManager().getModule(module).setEnabled(enabled);
+        if (!EP_LOADED) return;
+        ExploitPreventerCompat.disableAll();
     }
 }

--- a/src/main/java/anticope/rejects/utils/ExploitPreventerCompat.java
+++ b/src/main/java/anticope/rejects/utils/ExploitPreventerCompat.java
@@ -1,0 +1,35 @@
+package anticope.rejects.utils;
+
+import com.nikoverflow.exploitpreventer.ExploitPreventer;
+import com.nikoverflow.exploitpreventer.module.Modules;
+
+public class ExploitPreventerCompat {
+    public static void applyTranslationKey(boolean enabled) {
+        apply(Modules.TRANSLATION_KEY, enabled);
+    }
+
+    public static void applyFingerprinting(boolean enabled) {
+        apply(Modules.FINGERPRINTING, enabled);
+    }
+
+    public static void applyLocalHTTPRequest(boolean enabled) {
+        apply(Modules.LOCAL_HTTP_REQUEST, enabled);
+    }
+
+    public static void applyAll(boolean translationKey, boolean fingerprinting, boolean localHTTPRequest) {
+        apply(Modules.TRANSLATION_KEY, translationKey);
+        apply(Modules.FINGERPRINTING, fingerprinting);
+        apply(Modules.LOCAL_HTTP_REQUEST, localHTTPRequest);
+    }
+
+    public static void disableAll() {
+        apply(Modules.TRANSLATION_KEY, false);
+        apply(Modules.FINGERPRINTING, false);
+        apply(Modules.LOCAL_HTTP_REQUEST, false);
+    }
+
+    private static void apply(Modules module, boolean enabled) {
+        if (ExploitPreventer.getAPI() == null) return;
+        ExploitPreventer.getAPI().getModuleManager().getModule(module).setEnabled(enabled);
+    }
+}

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -39,5 +39,8 @@
     "java": ">=21",
     "minecraft": "~${mc_version}",
     "meteor-client": "*"
+  },
+  "suggests": {
+    "exploitpreventer": "*"
   }
 }

